pkg/scaffold/resource/resource.go

```diff
@@ -96,9 +99,16 @@ const (
 
 	// dns1123SubdomainMaxLength is a subdomain's max length in DNS (RFC 1123)
 	dns1123SubdomainMaxLength int = 253
+
+	dns1035LabelFmt    string = "[a-z]([-a-z0-9]*[a-z0-9])?"
```

Comment:
Shows that it is not the same validation for Kinds. Could we not have a kind == AppServer for example?

Reply:
Yep; the dns1035-like validation happens after the `Kind` [has been lowercased](https://github.com/kubernetes-sigs/kubebuilder/blob/5742ddd7a096570cd7fcc0881176d9dcb189d27b/pkg/scaffold/resource/resource.go#L83), so all of these combinations would be supported. apimachinery follows the same pattern, as far as I can tell (I'll find it and link here in a moment).

Reply:
Hi @mariantalla, 

See [here](https://github.com/kubernetes/apimachinery/blob/6e68a40eebf94cb3e778d807bce5637d6aa079ea/pkg/apis/meta/v1/types_swagger_doc_generated.go#L57-L58) that  kubernetes API andd apimachinery will allow Kinds with upper case.

The validation that you got from there is not used to check Kinds.  You can use this one [here](https://github.com/kubernetes/apimachinery/blob/ced427e1ea5fdd2d307fa083f960961ceff71b65/pkg/util/validation/validation.go#L123 ) to verify the kinds.

Reply:
I think @mariantalla means that apimachinery validates that the lower-case form of any kind is a valid dns-like label

Reply:
(although we should have links in comments referencing that)

Reply:
the relevant section: 

https://github.com/kubernetes/kubernetes/blob/b3b15a9e260ac65edfa14dd80add9046f3fa9b1f/staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go#L549-L582

@mariantalla is correct

Reply:
HI @mariantalla and @DirectXMan12,
 
Really tks for the explanations. I got that the lower case is just for the check and also, I tested locally the PR and all is working fine. It all shows fine, just the tests that were removed that IHMO should be back. We should still be ensuring that Crew and Guestbook and etc are accepted.

Reply:
Added the link in a few places in the conversation so that future-us can find it too :)